Products/ZenCollector/daemon.py

```diff
@@ -359,6 +370,25 @@ def generateEvent(self, event, **kw):
                 eventCopy['device_guid'] = guid
         return eventCopy
 
+    def should_trace_metric(self, metric, contextkey):
+        """
+        Tracer implementation - use this function to indicate whether a given
+        metric/contextkey combination is to be traced.
+        :param metric: name of the metric in question 
+        :param contextkey: context key of the metric in question
+        :return: boolean indicating whether to trace this metric/key
+        """
+        tests = []
+        if self.options.traceMetricName:
+            tests.append((self.options.traceMetricName, metric))
+        if self.options.traceMetricKey:
+            tests.append((self.options.traceMetricKey, contextkey))
+
+        result = [bool(re.compile(exp).match(subj)) for exp, subj in tests]
```

Comment:
If you are going to compile the expression for efficiency it should be done only once.